### PR TITLE
[ti] extract threat intel to helper

### PR DESCRIPTION
**Pull request by ryandeivert**
to @jacknagz or @mime-frame 
cc @airbnb/streamalert-maintainers 

### From @chunyong-lin :
* This refactoring will help users have more control when they use Threat Intel feature. For example, users can whitelist IOCs if they are false positive.

### From @ryandeivert :
* Small fix for pylint warnings and making `streamalert:ioc` a constant on the `StreamThreatIntel` class.

### CODE REVIEW

File: helpers/base.py [hunk 1]
```diff
@@ -110,3 +112,35 @@ def fetch_values_by_datatype(rec, datatype):
...
+        return False
+
+    for datatype in rec['normalized_types']:
+        if datatype not in datatypes_ioc_mapping:
+            continue
+        results = fetch_values_by_datatype(rec, datatype)
+        for result in results:
+            if (intel.get(datatypes_ioc_mapping[datatype])
+                    and result in intel[datatypes_ioc_mapping[datatype]]):
+                if 'streamalert:ioc' in rec:
```
Comment (jacknagz) [inline helpers/base.py]: side note: `streamalert:ioc` should probably be a const

Comment (ryandeivert) [reply helpers/base.py]: Addressed with: https://github.com/airbnb/streamalert/pull/327/commits/4296f6400cc43809cbf7b4bfbde4b678d0dc1a09

**Comment by mime-frame** [comment]
image

**Comment by ryandeivert** [comment]
@jacknagz rebased and made that a constant. @mime-frame seriously debating deleting your comment...

**Pull request merged**